### DISCUSSION TITLE
Problem viewing an IFC of a tunnel with Bonsai

### DISCUSSION THREAD

**Discussion by BimETS** [IfcOpenShell/IfcOpenShell#5538] [Q&A]
When opening the IFC file of a tunnel, the elements appear misaligned, and it shows a duplicate of other elements' geometry. If I close and reopen it, the same issue occurs, but the incorrect elements change. In BIMVision, it displays correctly. I'm not sure where the error is.
![Bonsai](https://github.com/user-attachments/assets/b5b45a32-53ec-48dd-930f-b9e5d7242278)
![BimVision](https://github.com/user-attachments/assets/a2a63d35-1b5a-43a0-a35a-b626a38dabea)

**Comment by aothms** [comment]
> If I close and reopen it, the same issue occurs, but the incorrect elements change

Never seen anything like this. Is this an IFC4.3 model? Can you share it? here, publicly, or confidentially at: models@ifcopenshell.org

**Reply by BimETS** [reply]
Thank you very much. I’ve sent it to you confidentially through a link since the files are quite large.

**Comment by aothms** [comment]
Hi, I tried a couple of times to reproduce this (IfcConvert, Bonsai, with/without multi-threading, old/new version of the code) but didn't spot anything peculiar.

What version are you using exactly, or did you try the very latest? Are these elements spurious copies of each other or are other elements disappearing?

**Reply by Moult** [reply]
If it helps, you can share the model at https://forms.gle/KMdbg1zBgeDRwtCK6 and that way I can help test too (and @Andrej730 will have access to it too)

**Reply by RickBrice** [reply]
Loop me in if the issues appear to be related to alignment geometry

**Reply by Moult** [reply]
@RickBrice I'm toying with the idea of using this Google forms to collect private files so that multiple devs can access it, it's searchable (we can search a spreadsheet of issue numbers and attached files), and then we can delete it when the issue is resolved. Would you like access to it? If so I think I'll need your email address (you can send to dion@thinkmoult.com if private).

**Comment by BimETS** [comment]
Thank you all very much, 
I tell you, the version I use in Blender is 4.2.0 and in Bonsai 0.8.0. I have tried again today...and it opens again as it wants the element. Attached screenshot.

![image](https://github.com/user-attachments/assets/bb85573f-e780-4d10-894d-d34a790c2376)

I don't understand why it doesn't give you problems.

 @Moult  I've sent you the file by the way you indicated.

**Reply by Moult** [reply]
I can recreate your issue. I'll investigate further tomorrow.

**Reply by Moult** [reply]
This is now fixed, it was a bug in Bonsai's native mesh handling. In the past, we would detect dense meshes and bypass IfcOpenShell's iterator because OCC was slow. No-doubt the implementation had some obscure bugs and didn't quite match the iterator's behaviour. Now with CGAL this workaround can be removed and also fixes your issue in the process.

**Reply by BimETS** [reply]
Thank you very much!!!! So, in order to see it correctly, do I have to activate something?

**Comment by BimETS** [comment]
I don't understand anything, it does this to me systematically every time I open it.

(https://github.com/user-attachments/assets/8de123a4-6933-46cc-9071-148dd4c9907f)

**Comment by BimETS** [comment]
A fact I don't know how important it is: I use Blender in portable format.

**Comment by Moult** [comment]
You can install the latest unstae build with these instructions https://docs.bonsaibim.org/guides/development/installation.html#unstable-installation

Dion Moult

Sent from Proton Mail mobile

-------- Original Message --------

**Reply by BimETS** [reply]
It works perfectly, thanks a lot.
